principal.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright (C)  2020 Yahoo Japan Corporation Athenz team.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package authorizerd
+
+// A Principal is an authenticated entity
+type Principal interface {
+	Name() string
+	Roles() []string
+	Domain() string
+	IssueTime() int64
+	ExpiryTime() int64
+}
+
+// A OAuthAccessToken is an interface for a principal that has a OAuthAccessToken
+type OAuthAccessToken interface {
+	ClientID() string
+}
+
+type principal struct {
+	name       string
+	roles      []string
+	domain     string
+	issueTime  int64
+	expiryTime int64
+}
+
+type oAuthAccessToken struct {
+	principal
+	clientID string
+}
+
+// Return the principal's name
```

Comment:
```suggestion
// Name returns the principal's name
```
Just to be consistent with other comments, `<function name> <verb> <description>`
Please also help to change the comment of other getters.